File: framework/languages/java/samples/java/ShoppingMallCheckout/Server/CommerceApi/src/main/java/systems/zlink/samples/shoppingmallcheckout/server/commerceapi/CommerceApiInstanceOptions.java
package systems.zlink.samples.shoppingmallcheckout.server.commerceapi;

import systems.zlink.samples.shoppingmallcheckout.server.configuration.SampleNames;

/** Parsed {@code --instance} selector for a CommerceApi process. */
public record CommerceApiInstanceOptions(String instanceId) {
    public static CommerceApiInstanceOptions fromArgs(String[] args) {
        String instanceId = SampleNames.ApiInstanceA;
        for (int i = 0; i < args.length - 1; i++) {
            if ("--instance".equals(args[i])) {
                instanceId = args[i + 1];
            }
        }
        return new CommerceApiInstanceOptions(instanceId);
    }

    public static String peerInstance(String instanceId) {
        return SampleNames.ApiInstanceB.equals(instanceId)
            ? SampleNames.ApiInstanceA
            : SampleNames.ApiInstanceB;
    }
}

File: framework/languages/java/samples/kotlin/ShoppingMallCheckout/Server/CommerceApi/src/main/kotlin/systems/zlink/samples/kotlin/shoppingmallcheckout/server/commerceapi/CommerceApiInstanceOptions.kt
package systems.zlink.samples.kotlin.shoppingmallcheckout.server.commerceapi

import systems.zlink.samples.kotlin.shoppingmallcheckout.server.configuration.SampleNames

/** Parsed `--instance` selector for a CommerceApi process. */
data class CommerceApiInstanceOptions(val instanceId: String) {
    companion object {
        fun fromArgs(args: Array<String>): CommerceApiInstanceOptions {
            var instanceId = SampleNames.ApiInstanceA
            for (i in 0 until args.size - 1) {
                if (args[i] == "--instance") {
                    instanceId = args[i + 1]
                }
            }
            return CommerceApiInstanceOptions(instanceId)
        }

        fun peerInstance(instanceId: String): String =
            if (instanceId == SampleNames.ApiInstanceB) SampleNames.ApiInstanceA else SampleNames.ApiInstanceB
    }
}
